core/code/ornaments.js

```diff
@@ -13,52 +13,120 @@
 // - Beacons (pe$TAG - $NAME) ie: 'peNIA - NIANTIC'
 // - Frackers ('peFRACK')
 // (there are 7 different colors for each of them)
+//
+// Ornament IDs are dynamic. NIANTIC might change them at any time without prior notice.
+// New ornamnent IDs found on the map will be recorded and saved to knownOrnaments from
+// which the Ornaments dialog will be filled with checked checkboxes.
+// To exclude a set of ornaments, even if they have not yet shown up on the map, the user
+// can add an entry to excludedOrnaments, which will compared (startsWith) to all known and
+// future IDs. example: "ap" to exclude all Ornaments for anomalies (ap1, ap2, ap2_v)
 
+/* global L, dialog */
 
 window.ornaments = {
 
   OVERLAY_SIZE: 60,
   OVERLAY_OPACITY: 0.6,
+  icon:[],
+  excludedOrnaments: [],
+  knownOrnaments: {},
 
   setup: function () {
     this._portals = {};
-    var layerGroup = L.layerGroup;
+    this.layerGroup = L.layerGroup;
     if (window.map.options.preferCanvas && L.Browser.canvas && !window.DISABLE_CANVASICONLAYER) {
-      layerGroup = L.canvasIconLayer;
+      this.layerGroup = L.canvasIconLayer;
       L.CanvasIconLayer.mergeOptions({ padding: L.Canvas.prototype.options.padding });
     }
-    this._layer = layerGroup();
-    this._beacons = layerGroup();
-    this._frackers = layerGroup();
-    window.layerChooser.addOverlay(this._layer, 'Ornaments');
-    window.layerChooser.addOverlay(this._beacons, 'Beacons');
-    window.layerChooser.addOverlay(this._frackers, 'Frackers');
+    this.load();
+
+    this.layers = {};
+    this.layers._layer = window.ornaments.layerGroup();
+    this.layers._excluded = window.ornaments.layerGroup(); // to keep excluded ornaments in an own layer
```

Comment:
no needs to use undescored names, you can directly use `this.layers` as a mapping object as it's used for the other layer IDs.
```suggestion
    this.layers = {};
    this.layers['Ornaments'] = window.ornaments.layerGroup();
    this.layers['Excluded ornaments'] = window.ornaments.layerGroup(); // to keep excluded ornaments in an own layer
```

Reply:
_layer is legacy. other, (3rd Party) plugins might be using it?

Reply:
you are right, but `this._layer` was legacy, this is `this.layers._layer` that's entirely new

Reply:
should we then have `_layer` redirect to `layers.['Ornaments']`

Reply:
yes, but what about legacy `._beacons` and `._frackers`?
core will break this anyway, maybe create them in the plugins or in `.createLayer` if called with `'Beacons'` or `'Frackers'`

Reply:
done